src/otx/core/model/entity/base.py

```diff
@@ -87,6 +88,11 @@ def forward(
     ) -> T_OTXBatchPredEntity | OTXBatchLossEntity:
         """Model forward function."""
         # If customize_inputs is overrided
+        if isinstance(inputs, TileBatchDetDataEntity):
+            return self.unpack_det_tiles(inputs)
+        if isinstance(inputs, TileBatchInstSegDataEntity):
+            return self.unpack_inst_seg_tiles(inputs)
```

Comment:
It feels difficult to me. This is because `OTXModel` is the base class. However, to deal with this if-branching, `OTXModel` is doing like that it has knowledge about the specifics of the derived classes. It is not desired in the perspective of object oriented programming.

Can we fix this problem such as

```python
    def forward(
        self,
        inputs: T_OTXBatchDataEntity,
    ) -> T_OTXBatchPredEntity | OTXBatchLossEntity:
        """Model forward function."""
        # If customize_inputs is overrided
        if isinstance(inputs, TileBatchDataEntity): # Common tile batch data entity
            return self.forward_tiles(inputs)

class DerivedOTXModel:
     def forward_tiles(...):
           ...
```

Reply:
Thanks for the suggestion, I added a common `OTXTileBatchDataEntity` and made changes accordingly
https://github.com/openvinotoolkit/training_extensions/pull/2764/files#diff-cb5e8239808876a102af23ebf579364e24835dd1e12cf6970f6da70e8c27be51R48